Restrict access to the admin password file

CAFxX opened the pull request:
The admin password file was previously using the default owner/permissions of `/var/vcap/store`: `root:root` and `0644`. Given the sensitive nature of its contents, change to `root:root` and `0600`.

Also ensure that all changes have been persisted to disk at the end of the setup script (otherwise a host crash immediately after the execution of the setup script could have left the job in a broken state - e.g. if the admin_password_file hadn't been synced to disk).

Tested both upgrading from an existing deployment and creating a new deployment:
```
influxdb/e5667339-ba90-4e9c-ad86-982b1bd3ed51:~$ ls -al /var/vcap/store/influxdb/
total 24
drwxr-xr-x 5 vcap vcap 4096 Jun 20 04:20 .
drwxr-xr-x 4 root root 4096 Jun 20 04:20 ..
-rw------- 1 root root   33 Jun 20 04:20 admin_password_file
drwxr-xr-x 3 vcap vcap 4096 Jun 20 04:20 data
drwxr-xr-x 2 vcap vcap 4096 Jun 20 04:21 meta
drwx------ 3 vcap vcap 4096 Jun 20 04:20 wal
```

Commit:
Restrict access to the admin password file

Also ensure that all changes have been persisted to disk at the end of the setup script

giner commented on jobs/influxdb/templates/bin/influxdb_setup.erb:
```diff
@@ -71,10 +71,14 @@ INFLUXDB_HTTP_PORT=<%= p('influxdb.http.port') %>
 fi
-ADMIN_PASSWORD=$(cat "$ADMIN_PASSWORD_FILE")
+if [ $(stat -c %U "$ADMIN_PASSWORD_FILE") != "root" ]; then # TODO: merge with the previous if
+  sudo chown root:root "$ADMIN_PASSWORD_FILE"
```
What about moving this out to a new script and pass here as environment variable so that sudo is not required?

CAFxX commented on jobs/influxdb/templates/bin/influxdb_setup.erb:
```diff
@@ -71,10 +71,14 @@ INFLUXDB_HTTP_PORT=<%= p('influxdb.http.port') %>
+if [ $(stat -c %U "$ADMIN_PASSWORD_FILE") != "root" ]; then # TODO: merge with the previous if
+  sudo chown root:root "$ADMIN_PASSWORD_FILE"
+  sudo chmod 600 "$ADMIN_PASSWORD_FILE"
+fi
```
the separate `if` is required to migrate existing deployments. once all deployments have migrated the two `if`s can be merged.

b10s reviewed (approved):
LGTM

Commit:
Generate the password directly from /dev/urandom

This avoids the openssl error `unable to write 'random state'`

Also drop some useless `sudo`

Commit:
Do not chown -r /var/vcap/store/influxdb

b10s commented on jobs/influxdb/templates/bin/influxdb_setup.erb:
```diff
@@ -72,11 +72,11 @@ ADMIN_USERNAME="<%= admin_username %>"
-sudo chmod 0600 "$ADMIN_PASSWORD_FILE"
-sudo chown root:root "$ADMIN_PASSWORD_FILE"
-ADMIN_PASSWORD=$(sudo cat "$ADMIN_PASSWORD_FILE")
+chmod 0600 "$ADMIN_PASSWORD_FILE"
```
why did you remove sudo?

b10s commented on jobs/influxdb/templates/bin/influxdb_setup.erb:
```diff
@@ -72,11 +72,11 @@ ADMIN_USERNAME="<%= admin_username %>"
 if [ ! -f "$ADMIN_PASSWORD_FILE" ]; then
   # generate the admin password and store it in /var/vcap/store
-  openssl rand -hex 16 > "$ADMIN_PASSWORD_FILE"
+  xxd -l 16 -pu </dev/urandom >"$ADMIN_PASSWORD_FILE"
```
Looks likes `u` here is useless:

```
$ xxd -l 16 -pu /dev/urandom
2770fa33e13d46973237ce69ea5beaa1
```

probably it was supposed to be:

```
$ xxd -u -l 16 -p /dev/urandom
C87E178B1E4448EF0053023A106F83D2
```

or without `u`

b10s commented on jobs/influxdb/templates/bin/influxdb_ctl.erb:
```diff
@@ -19,12 +19,18 @@ exec 2> >( tee -a /var/vcap/sys/log/influxdb_ctl.err.log | logger -p user.error
 ulimit -c unlimited
 ulimit -n <%= p("influxdb.max_open_fd") %>
 
-for DIR in $DATA_DIR $STORE_DIR $LOG_DIR $RUN_DIR; do
```
Looks like now it is going not only no `chown` but also no `mkdir` and no `chmod`.
Is it expected?

b10s commented on jobs/influxdb/templates/bin/influxdb_ctl.erb:
```diff
@@ -19,12 +19,18 @@ exec 2> >( tee -a /var/vcap/sys/log/influxdb_ctl.err.log | logger -p user.error
 ulimit -n <%= p("influxdb.max_open_fd") %>
 
-for DIR in $DATA_DIR $STORE_DIR $LOG_DIR $RUN_DIR; do
+for DIR in $DATA_DIR $LOG_DIR $RUN_DIR; do
```
Why we do not need to `chown -R` `/var/vcap/store/influxdb`  ?

CAFxX replied:
because if you do it recursively it overwrites the owner/permissions of /var/vcap/store/influxdb/admin_password_file that as the PR description says, should be root:root/0600 (instead of vcap:vcap/755)
